### DISCUSSION TITLE
How to set file permission while encrypting a file

### DISCUSSION THREAD

**Discussion by TechStackGit** [MatthiasValvekens/pyHanko#191] [Q&A]
Here I have a question about encrypting a file with set some file permissions like (`PDF_PERM_ACCESSIBILITY, PDF_PERM_PRINT, PDF_PERM_COPY, PDF_PERM_PRINT_HQ`). But I can't achieve it in pyHanko and here I attached my code for reference. Thank in advance.

```

from pyhanko.pdf_utils.reader import PdfFileReader 
from pyhanko.pdf_utils.writer import copy_into_new_writer 


with open("original.pdf",'rb') as inf:
    r = PdfFileReader(inf) 
    w = copy_into_new_writer(r) 
    w.encrypt(owner_pass='test',user_pass=None, **{"perms": -2580})
    with open(f"test1.pdf", 'wb') as outf:
        w.write(outf)
    outf.close()
inf.close()
```

**Comment by MatthiasValvekens** [comment]
Hi @TechStackGit,

Your code should work, except for the fact that specifying `user_pass=None` will cause the user password to be set to the same value as the owner password. Applications like Acrobat Reader will not enforce permissions when a file is opened with the owner password, which might explain the behaviour you're seeing. Try this instead:

```python
w.encrypt(owner_pass='test', user_pass='', perms=-2580)
```

Alternatively, you can set the `owner_pass` to some random value and `user_pass` to the password you actually want.

Also, in the spirit of transparency: these permissions and distinctions between different kinds of passwords are pretty much security theatre. They influence the UX a bit in graphical viewers, but in reality anyone with access to either the owner or user password can decrypt your file and/or remove the permissions no matter what. They're really only a barrier to non-technical users.
